src/client/VZCodeContext.tsx

```diff
@@ -172,6 +178,52 @@ export const VZCodeProvider = ({
     setUsername,
   } = useActions(dispatch);
 
+  // initialize open tabs from the url search parameters
+  useEffect(() => {
+    const search = new URLSearchParams(
+      document.location.search,
+    );
+    if (search.size === 0) {
+      return;
+    }
+    const tabList: TabState[] = [];
+    let activeFileId = null;
+    for (const [fileId, tabState] of Array.from(
+      search.entries(),
+    )) {
+      const isTransient = tabState.includes('t');
```

Comment:
Looks like a clever encoding scheme. Ideally we could have some unit tests for encoding and parsing of these.